src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -92,30 +88,27 @@ public void deliveryComplete(IMqttDeliveryToken token) {
      * @throws MQTTClientException if unable to create client for the mqtt broker
      */
     @Inject
-    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService)
-            throws MQTTClientException {
+    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+                      ExecutorService executorService) throws MQTTClientException {
         this(topics, mqttClientKeyStore, executorService, null);
         // TODO: Handle the case when serverUri is modified
         try {
-            this.mqttClientInternal = new MqttClient(serverUri, clientId, dataStore);
+            this.mqttClientInternal = new MqttClient(brokerUri.toString(), clientId, dataStore);
         } catch (MqttException e) {
             throw new MQTTClientException("Unable to create an MQTT client", e);
         }
     }
 
-    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService,
-                         IMqttClient mqttClient) {
-        // serverUri should take precedence since brokerServerUri is deprecated
-        String tmpUri = Coerce.toString(
-                topics.findOrDefault(DEFAULT_BROKER_URI, KernelConfigResolver.CONFIGURATION_CONFIG_KEY,
-                        DEPRECATED_BROKER_URI_KEY));
-        this.serverUri = Coerce.toString(
-                topics.findOrDefault(tmpUri, KernelConfigResolver.CONFIGURATION_CONFIG_KEY,
-                        BROKER_URI_KEY));
+    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+                         ExecutorService executorService, IMqttClient mqttClient) throws MQTTClientException {
+        try {
+            this.brokerUri = BridgeConfig.getBrokerUri(topics);
```

Comment:
if the URI changes are you going to recreate this whole MQTTClient instance? That's probably a reasonable solution if that's you plan.

Reply:
haven't thought through it completely yet, but yeah, having bridge component recreate this seems appropriate.

Reply:
update:  MQTTClient is now recreated on configuration change